components/CommentsApi.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react'
+import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
+import UnfoldMoreIcon from '@mui/icons-material/UnfoldMore';
+
+
+export default class CommentsApi extends Component {
+  render() {
+    const comments = this.props.comments;
```

Comment:
don't create variable in render , as this.props.comments will again and again assign into comments variable,
directly use this.props.comments in place of comments variable

Reply:
updated whole code with all the required optimization